aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -1180,5 +1181,52 @@ Tensor _reshape_nested(const Tensor& self, IntArrayRef proposed_shape) {
   return output;
 }
 
+Tensor slice_nested(
+    const Tensor& self,
+    int64_t dim,
+    c10::optional<int64_t> start,
+    c10::optional<int64_t> end,
+    int64_t step) {
+  int64_t ndim = self.dim();
+  if (ndim == 0) {
+    TORCH_CHECK_INDEX(false, "slice() cannot be applied to a 0-dim tensor.");
+  }
+  dim = maybe_wrap_dim(dim, ndim);
+  TORCH_CHECK(dim > 0, "Nested tensor dimension 0 cannot be sliced");
+
+  // will error if the dimension is jagged
+  auto dim_size = self.size(dim);
```

Comment:
I think you could also do opt_size[dim] == -1 and then provide a more collocated error message but thats a little nitty

Reply:
True, but the error message isn't too bad :p